pkg/controllers/dynakube/monitoredentities/reconciler.go

```diff
@@ -18,38 +20,48 @@ func NewReconciler( //nolint
 	dk *dynakube.DynaKube,
 ) controllers.Reconciler {
 	return &Reconciler{
-		dk:       dk,
-		dtClient: dtClient,
+		dk:           dk,
+		dtClient:     dtClient,
+		timeProvider: timeprovider.New(),
 	}
 }
 
 type Reconciler struct {
-	dk       *dynakube.DynaKube
-	dtClient dynatrace.Client
+	dk           *dynakube.DynaKube
+	dtClient     dynatrace.Client
+	timeProvider *timeprovider.Provider
 }
 
 func (r *Reconciler) Reconcile(ctx context.Context) error {
-	if r.dk.Status.KubernetesClusterMEID == "" {
-		log.Info("start reconciling monitored entities")
+	log.Info("start reconciling monitored entities")
 
-		monitoredEntities, err := r.dtClient.GetMonitoredEntitiesForKubeSystemUUID(ctx, r.dk.Status.KubeSystemUUID)
-		if err != nil {
-			log.Info("failed to retrieve MEs")
+	if !conditions.IsOutdated(r.timeProvider, r.dk, MEIDConditionType) {
+		log.Info("Kubernestes Cluster MEID not outdated, skipping reconciliation")
 
-			return err
-		}
+		return nil
+	}
 
-		if len(monitoredEntities) == 0 {
-			log.Info("no MEs found, no kubernetesClusterMEID will be set in the dynakube status")
+	conditions.SetStatusOutdated(r.dk.Conditions(), MEIDConditionType, "Kubernetes Cluster MEID is outdated in the status")
 
-			return nil
-		}
+	monitoredEntities, err := r.dtClient.GetMonitoredEntitiesForKubeSystemUUID(ctx, r.dk.Status.KubeSystemUUID)
+	if err != nil {
+		log.Info("failed to retrieve MEs")
 
-		r.dk.Status.KubernetesClusterMEID = findLatestEntity(monitoredEntities).EntityId
+		return err
+	}
 
-		log.Info("kubernetesClusterMEID set in dynakube status, done reconciling", "kubernetesClusterMEID", r.dk.Status.KubernetesClusterMEID)
+	if len(monitoredEntities) == 0 {
+		log.Info("no MEs found, no kubernetesClusterMEID will be set in the dynakube status")
+		setEmptyMEIDCondition(r.dk.Conditions())
+
+		return nil
 	}
 
+	r.dk.Status.KubernetesClusterMEID = findLatestEntity(monitoredEntities).EntityId
+	conditions.SetStatusUpdated(r.dk.Conditions(), MEIDConditionType, "Kubernestes Cluster MEID is up to date in the status: "+r.dk.Status.KubernetesClusterMEID)
```

Comment:
Would create a shorter message here, like "Kubernestes Cluster MEID is up to date" only

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/3541/commits/f02da5dac342c1f5adb3818ea08fc186560da86e